src/mapdata.h

```diff
@@ -447,6 +447,8 @@ struct submap {
     trap_id         trp[SEEX][SEEY];  // Trap on each square
     int             rad[SEEX][SEEY];  // Irradiation of each square
 
+    bool            known[SEEX][SEEY]; // has the player ever seen this tile
```

Comment:
You can save the game and start a new one with a different character in the very same world.

There is also the possibility of map sharing, see http://smf.cataclysmdda.com/index.php?topic=9156, which usually means several human players use the same map data.